torchao/_models/llama/generate.py

```diff
@@ -229,9 +229,39 @@ def main(
             groupsize=int(quantization.split("-")[1])
             assert groupsize in [32,64,128,256], f"int4wo groupsize needs to be one of [32,64,128,256] but got {groupsize}"
             quantize_(model, int4_weight_only(group_size=groupsize))
-        if "marlin" in quantization:
+        if "sparse-marlin" in quantization:
             from torchao.dtypes import MarlinSparseLayout
             quantize_(model, int4_weight_only(layout=MarlinSparseLayout()))
+        if "gptq-marlin" in quantization:
+            # avoid circular imports
+            from torchao._models._eval import InputRecorder
+            from torchao.quantization.GPTQ import Int4WeightOnlyGPTQQuantizer
+
+            group_size = 128
+            pad_calibration_inputs = True
+            calibration_tasks = ["wikitext"]
+            calibration_limit = 1
+
+            assert precision == torch.float16, f"{quantization} requires precision or float16 but got {precision}"
+            assert "cuda" in device, "gptq sparse marlin quantization only works on cuda"
+            assert torch.cuda.get_device_capability()[0] >= 8, "gptq sparse marlin quantization only works on cuda with compute capability >= 8"
+
+            inputs = InputRecorder(
```

Comment:
I think you are basing your code off the old GPTQ not the new GPTQ, see https://github.com/pytorch/ao/blob/main/test/quantization/test_quant_api.py#L343